modelBuilding/model-building-methodology.py

```diff
@@ -0,0 +1,247 @@
+'''
+Notes by Kimathi, so hit me up if I don't make any sense
+Acknowledgements: https://blog.keras.io/building-powerful-image-classification-models-using-very-little-data.html
+GENERAL OVERVIEW---------------------------------------------------------------
+- train a small network from scratch (as a baseline)
+- use the bottleneck features of a pre-trained network
+- fine-tune the top layers of a pre-trained network
+'''
+
+'''
+STEP: ORGANIZING DATASET-----------------------------------------------------
+1000 training examples per class, and 400 validation examples per class
+Data set directory stucture:
+    data/
+        train/
+            dogs/
+                dog001.jpg
+                dog002.jpg
+                ...
+            cats/
+                cat001.jpg
+                cat002.jpg
+                ...
+        validation/
+            dogs/
+                dog001.jpg
+                dog002.jpg
+                ...
+            cats/
+                cat001.jpg
+                cat002.jpg
+                ...
+'''
+
+'''
+STEP: DATA PRE-PROCESSING AND DATA AUGMENTATION------------------------------
+- Augment the data to increase sample size and prevent overfitting and helping
+  the model generalize better
+- In Keras this can be done via the ImageDataGenerator class allowing you to
+  - configure random transformation and normalization operations for training
+  - instantiate generations of augmented image batches (and their labels)
+    via .flow(data, labels) or .flow_from_directory(directory). These
+    generations can be used with the Keras model methods that accept data
+    generators as inputs, fit_generator, evaluate_generator and
+    predict_generator
+'''
+# Example----------------------------------------------------------------------
+from keras.preprocessing.image import ImageDataGenerator
+from keras.preprocessing.image import array_to_img, img_to_array, load_img
+datagen = ImageDataGenerator(
```

Comment:
Can we create a function which does this, rather than just the example code?

Reply:
Sure, I'll add that functionality